ocaml/xapi/xapi_vif.ml

```diff
@@ -116,3 +162,23 @@ let add_ipv6_allowed ~__context ~self ~value =
 let remove_ipv6_allowed ~__context ~self ~value =
 	change_locking_config ~__context ~self ~licence_check:false
 		(fun () -> Db.VIF.remove_ipv6_allowed ~__context ~self ~value)
+
+let add_to_static_ip_setting ~__context ~self ~key ~value =
+	change_locking_config ~__context ~self ~licence_check:false
+		(fun () -> 
+			assert_ip_setting_is "static_ip_setting" key value;
+			let vm = Db.VIF.get_VM ~__context ~self in
+			let vm_gm = Db.VM.get_guest_metrics ~__context ~self:vm in 
+			let network_optimized = try Db.VM_guest_metrics.get_network_paths_optimized ~__context ~self:vm_gm with _ -> false in
```

Comment:
This is where we should instead check a feature flag from the guest agent to assert that the agent support setting IP addresses.